perceval/backends/jenkins.py

```diff
@@ -0,0 +1,243 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import CacheError
+from ..utils import (DEFAULT_DATETIME,
+                     datetime_to_utc,
+                     str_to_datetime,
+                     urljoin)
+
+logger = logging.getLogger(__name__)
+
+class Jenkins(Backend):
+    """Jenkins backend for Perceval.
+
+    This class retrieves the builds from a
+    Jenkins site. To initialize this class the
+    site must be provided.
+
+    :param site: Jenkins site
+    :param job_url: Jenkins job from which to get the builds
+    :param cache: cache object to store raw data
+    :param origin: identifier of the repository; when `None` or an
+        empty string are given, it will be set to `site` value
+    """
+    version = '0.1.0'
+
+    def __init__(self, site, job_url=None, cache=None, origin=None):
+        origin = origin if origin else site
+
+        super().__init__(origin, cache=cache)
+        self.site = site
+        self.job_url = job_url
+        self.client = JenkinsClient(site)
+
+    @metadata
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        """Fetch the builds from the site.
+
+        The method retrieves, from a Jenkins site, the
+        builds updated since the given date.
+
+        :param from_date: Not supported
+
+        :returns: a generator of builds
+        """
+
+        logger.info("Looking for projects at site '%s'", self.site)
+
+        self._purge_cache_queue()
+        nbuilds = 0  # number of builds processed
+        njobs = 0 # number of jobs processed
+
+        if not self.job_url:
+            projects = json.loads(self.client.get_jobs())
+            jobs = projects['jobs']
+        else:
+            jobs = [{'url':self.job_url}]
+
+        for job in jobs:
+            njobs += 1
+            logger.debug("Adding builds from %s (%i/%i)" % (job['url'], njobs, len(jobs)))
+            raw_builds = self.client.get_builds(job['url'])
+            self._push_cache_queue(raw_builds)
+            self._flush_cache_queue()
+            builds = json.loads(raw_builds)
+            builds = builds['builds']
+            for build in builds:
```

Comment:
So, the main item we return here is builds. Why builds instead of jobs? I think it would make sense to return builds if  `from_date` was used but since it isn't used maybe it is better to return a job as an item.